### ISSUE TITLE
CSV to TF records, polygon label dataset- mask rcnn object detection

### ISSUE THREAD

**Issue by xyzdcgan** [matterport/Mask_RCNN#1547]
I am creating custom mask-rcnn object detection model.
I labeled the data with polygon shape and created csv file.

I am not able to create tf record from csv as, it has polygon set rather then rectangle bounding box.
Can you help in creating tf record..?

Also i am using this tensroflow, and using mask_rcnn_resnet101_atrous_coco from https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md

And configuring config file and giving path for it.

https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/training.html

Using this kind of implementation for mask-rcnn, so I can get checkpoint file.
Rather then using a pre trained h5 model of mask-rcnn

Thanks

**Comment by fevzidas** [comment]
Hello @xyzdcgan  do you have any solution for your problem? 

I have same problem.

**Comment by May-forever** [comment]
Dear xyzdcgan ,

I want to use Mask-RCNN to train on my custom dataset (just one class).

First of all, I use Labelme to create groundtruth for each image. Each image has a .json file as its ground-truth.

After that, I don't know what should I do...

I want to know how to convert my groundtruth file to a format that Mask-rcnn can use. Do you have any tools ?

Looking forward to hearing from you, thanks in advance.

With best regards,
Ming

**Comment by xyzdcgan** [comment]
Hi @fevzidas @hjm1990818 

steps i followed to use mask-rcnn:

1) Labelled each custom images with polygon
2) Save that files in json format
3) Used a pretrained model 
4) Trained that model with my images

It was working fine 

https://github.com/priya-dwivedi/Deep-Learning/tree/master/mask_rcnn_damage_detection

I used above link for all things to detect using mask-rcnn

**Comment by May-forever** [comment]
> Hi @fevzidas @hjm1990818
> 
> steps i followed to use mask-rcnn:
> 
> 1. Labelled each custom images with polygon

Dear xyzdcgan,

Thank you very much for your response.

Maybe you don't understand what I mean, I am not a English native.

Could you please give me your e-mail address? I want to show some examples for you for a detailed explaination.

Looking forward to hearing from you, thanks.

With best regards,
MIng

**Comment by xyzdcgan** [comment]
priyanksolanki900@gmail.com
